### PR TITLE
[Metrics] Continue implementation of metrics controller

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
- Continued wiring up the metrics controller with storage and data model objects
- Filled in some documentation now that things are becoming more stable

### Next up
- Implement metrics unit tests

### CODE REVIEW

File: GoogleDataTransport/GDTCORLibrary/Internal/GDTCORStorageProtocol.h [hunk 1]
```diff
@@ -123,7 +121,10 @@ typedef void (^GDTCORStorageBatchBlock)(NSNumber *_Nullable newBatchID,
 
 @end
 
-// TODO: Consider complete replacing block based API by promise API.
+// TODO(ncooke3): Consider complete replacing block based API by promise API.
```
Comment (paulb777) [inline GoogleDataTransport/GDTCORLibrary/Internal/GDTCORStorageProtocol.h:124]: Done now?

**Review by paulb777 — APPROVED** [review]

File: GoogleDataTransport/GDTCORLibrary/Internal/GDTCORStorageProtocol.h (hunk 1 above)
Comment (ncooke3) [reply GoogleDataTransport/GDTCORLibrary/Internal/GDTCORStorageProtocol.h:124]: No, this is not done (and not in scope for this metrics project). Resolving this TODO would mean removing the interface on the lines above it. 

I have a good idea of what needs to be done so I just added my username to it for future work.

File: GoogleDataTransport/GDTCCTTests/Common/TestStorage/GDTCCTTestStorage.m [hunk 2]
```diff
@@ -207,10 +207,14 @@ - (void)defaultBatchWithEventSelector:(nonnull GDTCORStorageEventSelector *)even
       }];
 }
 
-- (FBLPromise<GDTCORLibraryData> *)
-    fetchAndUpdateLibraryDataForKey:(NSString *)key
-                              klass:(Class)klass
-                     readWriteBlock:(GDTCORStorageLibraryDataReadWriteBlock)readWriteBlock {
+- (FBLPromise<NSNull *> *)fetchAndUpdateClientMetricsWithReadWriteBlock:
+    (GDTCORMetricsMetadata *_Nullable (^)(GDTCORMetricsMetadata *_Nullable fetchedMetadata,
+                                          NSError *_Nullable fetchError))readWriteBlock {
```
Comment (chliangGoogle) [inline GoogleDataTransport/GDTCCTTests/Common/TestStorage/GDTCCTTestStorage.m:212]: fetchAndUpdateClientMetricsWithHandler?

File: GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m [hunk 3]
```diff
@@ -114,22 +104,25 @@ - (instancetype)initWithStorage:(id<GDTCORStoragePromiseProtocol>)storage {
...
+      .then(^GDTCORMetrics *(NSArray *metricsMetadataAndStorageMetadata) {
+        return
+            [GDTCORMetrics metricsWithMetricsMetadata:metricsMetadataAndStorageMetadata.firstObject
+                                      storageMetadata:metricsMetadataAndStorageMetadata.lastObject];
       });
 }
 
 - (nonnull FBLPromise<NSNull *> *)offerMetrics:(nonnull GDTCORMetrics *)metrics {
-  GDTCORStorageLibraryDataReadWriteBlock readWriteblock = ^GDTCORMetricsMetadata *(
+  __auto_type readWriteblock = ^GDTCORMetricsMetadata *(
```
Comment (chliangGoogle) [inline GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m:120]: maybe rename it to fetchAndUpdateCompleteHandler to make it more consistent with function name and easier to understand

**Review by chliangGoogle — APPROVED** [review]

File: GoogleDataTransport/GDTCCTTests/Common/TestStorage/GDTCCTTestStorage.m (hunk 2 above)
Comment (ncooke3) [reply GoogleDataTransport/GDTCCTTests/Common/TestStorage/GDTCCTTestStorage.m]: Sure, I'll ensure it is fully documented.

Done in 1c37aa6

File: GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m (hunk 3 above)
Comment (ncooke3) [reply GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m]: Done in 1c37aa6

**Pull request merged**